### ISSUE TITLE
WinForms designer creates incorrect #region directives when creating new event handlers in single form.prg files

### ISSUE THREAD

**Issue by cpyrgas** [X-Sharp/XSharpPublic#935]
Problem happens when a form does not contain a separate .designer.prg file, but has all code in the form.prg. To reproduce:

- Create a new winforms project
- Add a new form with no separate designer.prg file, like in the attached solution
- Add any event to the form/controls in the form, for example double click on the form or add any other event, the generated code looks like:

```
    PUBLIC PARTIAL CLASS Form1	;
		INHERIT System.Windows.Forms.Form
/ /[....]
        PROTECTED METHOD Dispose(disposing AS LOGIC) AS VOID  STRICT

            IF (disposing .AND. (components != NULL))
                components:Dispose()
            ENDIF
            SUPER:Dispose(disposing)
			RETURN

        #region Windows Form Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        PRIVATE METHOD InitializeComponent() AS VOID STRICT
			SELF:SuspendLayout()
			//	
			//	Form1
			//	
			SELF:AutoScaleDimensions	:=	System.Drawing.SizeF{6, 13}
			SELF:AutoScaleMode	:=	System.Windows.Forms.AutoScaleMode.Font
			SELF:ClientSize	:=	System.Drawing.Size{525, 321}
			SELF:Name	:=	"Form1"
			SELF:Text	:=	"Form1"
			SELF:Load	+=	System.EventHandler{ SELF, @Form1_Load() }
			SELF:KeyDown	+=	System.Windows.Forms.KeyEventHandler{ SELF, @Form1_KeyDown() }
			SELF:KeyPress	+=	System.Windows.Forms.KeyPressEventHandler{ SELF, @Form1_KeyPress() }
			SELF:ResumeLayout(FALSE)
        #region Windows Form Designer generated code
        PRIVATE METHOD Form1_Load(sender AS System.Object, e AS System.EventArgs) AS VOID STRICT
			RETURN
        #region Windows Form Designer generated code
        PRIVATE METHOD Form1_KeyDown(sender AS System.Object, e AS System.Windows.Forms.KeyEventArgs) AS VOID STRICT
			RETURN
        #region Windows Form Designer generated code
        PRIVATE METHOD Form1_KeyPress(sender AS System.Object, e AS System.Windows.Forms.KeyPressEventArgs) AS VOID STRICT
			RETURN

        #endregion

    END CLASS 
```

So there's a new #region directive inserted for every event handler. And because of that, the project does not compile, the compiler complains about missing #endregion directives. After manually deleting the directives, it works fine, until adding a new event to the form.

**Comment by RobertvanderHulst** [comment]
This is already fixed in 2.12.1

**Comment by cpyrgas** [comment]
Confirmed fixed
